test/e2e/flask/petnames-transactions.spec.js

```diff
@@ -0,0 +1,48 @@
+const {
+  openDapp,
+  switchToNotificationWindow,
+  withFixtures,
+  unlockWallet,
+  defaultGanacheOptions,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const {
+  expectName,
+  focusTestDapp,
+  rejectSignatureOrTransactionRequest,
+  saveName,
+} = require('./petnames-helpers');
+
+async function createTransactionRequest(driver) {
+  await driver.clickElement('#sendButton');
+  await driver.delay(3000);
+}
+
+describe('Petnames', function () {
+  it('can save names for addresses in transaction requests', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withNoNames()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await createTransactionRequest(driver);
+        await switchToNotificationWindow(driver, 3);
+        await expectName(driver, '0x0c54F...7AaFb', false);
+        await saveName(driver, '0x0c54F...7AaFb', 'test.lens', undefined);
```

Comment:
In the signature tests, there were multiple addresses per confirmation, so for some we used a proposed name, and for one we saved a custom name.

Here we are exclusively using a proposed name so there is an argument that we could have a separate test for saving a transaction with a custom name, though it's low risk since we know it's fundamentally the same underlying component.

Equally we know that most transaction types ultimately use the `sender-to-recipient` component we've updated but for the sake of coverage, we could also add additional tests for different transaction types such as send transactions from within the wallet, or an approve transaction which uses some alternate components.

Reply:
Moved to #22515